### PR TITLE
Update architecture diagram and doc

### PR DESCRIPTION
**Pull request by alvinlin123**
Current architecture diagram still contains information about chunk store, this PR removes that. I also modify the architecture doc to have store gateway and compactor to be required services. 

I think the architecture document page is too lengthy, I plan to spend some more time later to only have summary of what each service does, and will extract the details to separate page. 

Signed-off-by: Alvin Lin <alvinlin@amazon.com>

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
image

Let's update or remove this `Text`?

File: docs/architecture.md [hunk 1]
```diff
@@ -170,6 +167,27 @@ Queriers fetch series samples both from the ingesters and long-term storage: the
 
 Queriers are **stateless** and can be scaled up and down as needed.
 
+### Compactor
+
+The **compactor** is an service which is responsible to:
```
Comment (friedrichg) [inline docs/architecture.md:172]:
```suggestion
The **compactor** is a service which is responsible to:
```

File: docs/architecture.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ slug: architecture
 
 Cortex consists of multiple horizontally scalable microservices. Each microservice uses the most appropriate technique for horizontal scaling; most are stateless and can handle requests for any users while some (namely the [ingesters](#ingester)) are semi-stateful and depend on consistent hashing. This document provides a basic overview of Cortex's architecture.
 
+The following digaram does not include all the Cortex services, but does represnet a typical deployment topology.
```
Comment (friedrichg) [inline docs/architecture.md:10]:
```suggestion
The following diagram does not include all the Cortex services, but does represent a typical deployment topology.
```

**Review by friedrichg — COMMENTED** [review]
minor nits

**Comment by alvinlin123** [comment]
> image
> 
> Let's update or remove this `Text`?

OMG, thanks for catching it...

**Review by yeya24 — APPROVED** [review]

**Review by friedrichg — APPROVED** [review]

**Pull request merged**